### PR TITLE
enable suspension for spammers

### PR DESCRIPTION
**Pull request by vpiyush**
Redefining p2p jailing to peer suspenstion. If a peer spams a node it would get suspended (will not be allowed to make connection ) for upto 60 blocks. This is the initial implementation.

### CODE REVIEW

File: p2p/server.go [hunk 1]
```diff
@@ -1061,9 +1067,11 @@ func (srv *Server) checkInboundConn(remoteIP net.IP) error {
 	return nil
 }
 
-func (srv *Server) AddJail(id enode.ID, jailTime time.Duration) {
-	now := srv.clock.Now()
-	srv.jailed.add(id.String(), now.Add(jailTime))
+func (srv *Server) needsSuspension(pd peerDrop) {
```
Comment (yazzaoui) [inline p2p/server.go:1070]: The current name suggests it checks if a suspension is needed, but from the code, it actually performs the suspension by adding a peer to a suspension list. I'd suggest to rename it to `processPeerSuspension`, `checkAndSuspendPeer` or something else.

**Review by yazzaoui — APPROVED** [review]
LGTM, would have preferred though using the more widespread word `blacklist`.

**Review by tbssajal — APPROVED** [review]
LGTM

**Comment by vpiyush** [comment]
> LGTM with naming comment, would have preferred though using the more widespread word `blacklist`.

The term "suspension" implies a temporary restriction, whereas "blacklist" conveys a more permanent prohibition, which might not be acurrate in this context. What do you think? I am open to rename it to blacklist though.

**Comment by yazzaoui** [comment]
>The term "suspension" implies a temporary restriction, whereas "blacklist" conveys a more permanent prohibition, which might not be acurrate in this context. What do you think? I am open to rename it to blacklist though.

Ok it's fine let's not quibble over it unless @cmjc you have a strong view on that ?

**Comment by cmjc** [comment]
No quibbling! lgtm Agree with @vpiyush .

**Pull request merged**